upcloud/kubernetes.go

```diff
@@ -0,0 +1,60 @@
+package upcloud
+
+import (
+	"fmt"
+)
+
+const (
+	KuberetesClusterStateConfiguring KubernetesClusterState = "configuring"
+	KuberetesClusterStateReady       KubernetesClusterState = "ready"
+	KubernetesClusterTypeStandalone  KubernetesClusterType  = "standalone"
+)
+
+type (
+	KubernetesClusterState string
+	KubernetesClusterType  string
+)
+
+type KubernetesCluster struct {
+	AutoUpgrade       *bool                       `json:"auto_upgrade"`
+	ClusterSubnet     string                      `json:"cluster_subnet"`
+	Description       string                      `json:"description"`
+	Labels            []Label                     `json:"labels"`
+	MaintenancePeriod KubernetesMaintenancePeriod `json:"maintenance_period"`
+	Name              string                      `json:"title"`
+	Network           string                      `json:"network"`
+	NodeGroups        []KubernetesNodeGroup       `json:"node_groups"`
+	State             KubernetesClusterState      `json:"state"`
+	StorageTemplateID string                      `json:"storage_template_id"` // TODO: needed and is it ID or UUID?
+	Type              KubernetesClusterType       `json:"type"`
+	UUID              string                      `json:"uuid"`
+	Version           string                      `json:"version"`
+	Zone              string                      `json:"zone"`
+}
+
+type KubernetesNodeGroup struct {
+	Count  *int    `json:"count,omitempty"`
+	Labels []Label `json:"labels,omitempty"`
+	Name   string  `json:"name,omitempty"`
+	Plan   string  `json:"plan,omitempty"`
+}
+
+type KubernetesMaintenancePeriod struct {
+	DOW  string `json:"dow,omitempty"`
+	Hour *int   `json:"hour,omitempty"`
+}
+type KubernetesPlan struct {
+	Name        string `json:"name"`
+	Description string `json:"description"`
+}
+
+type KubernetesError struct {
```

Comment:
Is this type needed or can we use `upcloud.Error` ?

Reply:
In the API spec there is field `error_details` (string) in the error schema, so created a new type for keeping things a bit cleaner. I can embed the `upcloud.Error`, but let's see if the details is needed in the end.